src/networking.c

```diff
@@ -1682,6 +1683,22 @@ void clientAcceptHandler(connection *conn) {
         }
     }
 
+    /* Auto-authenticate from cert_user field if set */
+    sds username = connGetPeerUsername(conn);
+    if (username != NULL) {
+        user *u = ACLGetUserByName(username, sdslen(username));
+        if (u) {
+            c->user = u;
+            c->flag.authenticated = true;
+            serverLog(LL_VERBOSE, "TLS: Auto-authenticated client as %s",
+                      server.hide_user_data_from_log ? "*redacted*" : u->name);
+        } else {
+            serverLog(LL_NOTICE, "TLS: Auto-authentication failed, user '%s' not found", username);
+            addACLLogEntry(c, ACL_INVALID_TLS_CERT_AUTH, 0, 0, username, sdsnew("User not found in auto TLS auth"));
```

Comment:
No need to store the string `"User not found in auto TLS auth"` in the ACL log. The new reason `ACL_INVALID_TLS_CERT_AUTH` already indicates this. This is a structured log, not a plain text verbose log.

The first 0 there (context) should be ACL_LOG_CTX_TOPLEVEL rather than the a magic number 0. The second 0 is an argument position so here 0 is fine. (Argpos is only used for some `reason` when the logging is triggered by some command.)

```suggestion
            addACLLogEntry(c, ACL_INVALID_TLS_CERT_AUTH, ACL_LOG_CTX_TOPLEVEL, 0, username, NULL);
```

Also, you need to handle this new reason when generating the reply of the `ACL LOG` command in `aclCommand()`, here: https://github.com/valkey-io/valkey/blob/8.1.2/src/acl.c#L3016. The output for this case can be something short like "tls-cert", which would match the other reasons.

You can view the log using the `ACL LOG` command. You should probably test this (at least manually in valkei-cli) to see how the reply looks after logging the new `ACL_INVALID_TLS_CERT_AUTH`.

Reply:
Btw, I reopened the resolved comment at serverLog (one line above this comment). Let's delete the line serverLog now that we have ACL log instead.

Reply:
Added suggested changes, also verified running locally :- 
```
127.0.0.1:6379> acl log
1)  1) "count"
    2) (integer) 1
    3) "reason"
    4) "tls-cert"
    5) "context"
    6) "toplevel"
    7) "object"
    8) ""
    9) "username"
   10) "client"
   11) "age-seconds"
   12) "48.788"
   13) "client-info"
   14) "id=2 addr=127.0.0.1:41126 laddr=127.0.0.1:6379 fd=10 name= age=0 idle=0 flags=N capa= db=0 sub=0 psub=0 ssub=0 multi=-1 watch=0 qbuf=0 qbuf-free=0 argv-mem=0 multi-mem=0 rbs=16384 rbp=0 obl=0 oll=0 omem=0 tot-mem=17024 events=r cmd=NULL user=default redir=-1 resp=2 lib-name= lib-ver= tot-net-in=0 tot-net-out=0 tot-cmds=0"
   15) "entry-id"
   16) (integer) 0
   17) "timestamp-created"
   18) (integer) 1751444186268
   19) "timestamp-last-updated"
   20) (integer) 1751444186268
```

Logs for ACL Stats :-
```
127.0.0.1:6379> INFO all
# Stats
total_connections_received:3
total_commands_processed:8
instantaneous_ops_per_sec:0
total_net_input_bytes:325
total_net_output_bytes:657476
total_net_repl_input_bytes:0
total_net_repl_output_bytes:0
instantaneous_input_kbps:0.00
instantaneous_output_kbps:0.00
instantaneous_input_repl_kbps:0.00
instantaneous_output_repl_kbps:0.00
rejected_connections:0
sync_full:0
sync_partial_ok:0
sync_partial_err:0
expired_keys:0
expired_stale_perc:0.00
expired_time_cap_reached_count:0
expire_cycle_cpu_milliseconds:2
evicted_keys:0
evicted_clients:0
evicted_scripts:0
total_eviction_exceeded_time:0
current_eviction_exceeded_time:0
keyspace_hits:0
keyspace_misses:0
pubsub_channels:0
pubsub_patterns:0
pubsubshard_channels:0
latest_fork_usec:0
total_forks:0
migrate_cached_sockets:0
slave_expires_tracked_keys:0
active_defrag_hits:0
active_defrag_misses:0
active_defrag_key_hits:0
active_defrag_key_misses:0
total_active_defrag_time:0
current_active_defrag_time:0
tracking_total_keys:0
tracking_total_items:0
tracking_total_prefixes:0
unexpected_error_replies:0
total_error_replies:1
dump_payload_sanitizations:0
total_reads_processed:12
total_writes_processed:15
io_threaded_reads_processed:0
io_threaded_writes_processed:0
io_threaded_freed_objects:0
io_threaded_accept_processed:0
io_threaded_poll_processed:0
io_threaded_total_prefetch_batches:0
io_threaded_total_prefetch_entries:0
client_query_buffer_limit_disconnections:0
client_output_buffer_limit_disconnections:0
reply_buffer_shrinks:3
reply_buffer_expands:0
eventloop_cycles:1815
eventloop_duration_sum:615377
eventloop_duration_cmd_sum:5522
instantaneous_eventloop_cycles_per_sec:9
instantaneous_eventloop_duration_usec:284
acl_access_denied_auth:0
acl_access_denied_cmd:0
acl_access_denied_key:0
acl_access_denied_channel:0
invalid_tls_cert_auth_failures:1
```